networkx/algorithms/approximation/broadcasting.py

```diff
@@ -0,0 +1,144 @@
+
+    The broadcast center of a graph G denotes the set of vertices having minimum broadcast time [1]_.
+    This is a linear algorithm for determining the broadcast center of a tree with N vertices,
+    as a by-product it can also determine the broadcast time from the broadcast center.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Undirected graph
+        The graph should be an undirected tree
+
+    Returns
+    -------
+    BC : (int, set) tuple
+        minimum broadcast number of the tree, set of broadcast centers
+
+    Raises
+    ------
+    NetworkXNotImplemented
+        If the graph is directed or is a multigraph.
+
+    References
+    ----------
+    .. [1] Slater, P.J., Cockayne, E.J., Hedetniemi, S.T,
+       Information dissemination in trees. SIAM J.Comput. 10(4), 692–701 (1981)
+    """
+    # Assert that the graph G is a tree
+    if not nx.is_tree(G):
+        NetworkXError("Your graph is not a tree")
+    # step 0
+    if G.number_of_nodes() == 2:
+        return 1, set(G.nodes())
+    elif G.number_of_nodes() == 1:
+        return 0, set(G.nodes())
+
+    # step 1
+    U = {node for node, deg in G.degree if deg == 1}
+    values = {n: 0 for n in U}
+    T = G.copy()
+    T.remove_nodes_from(U)
+
+    # step 2
+    W = {node for node, deg in T.degree if deg == 1}
+    values.update((w, G.degree[w] - 1) for w in W)
+
+    # step 3
+    while T.number_of_nodes() >= 2:
+        # step 4
+        w = min(W, key=lambda n: values[n])
+        v = next(T.neighbors(w))
+
+        # step 5
+        U.add(w)
+        W.remove(w)
+        T.remove_node(w)
+
+        # step 6
+        if T.degree(v) == 1:
+            # update t(v)
+            values.update({v: _get_max_broadcast_value(G, U, v, values)})
+            W.add(v)
+
+    # step 7
+    v = nx.utils.arbitrary_element(T)
+    b_T = _get_max_broadcast_value(G, U, v, values)
+    return b_T, _get_broadcast_centers(G, v, values, b_T)
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def tree_broadcast_time(G, node=None):
+    """Return the Broadcast Time of the tree G.
+
+    Broadcasting is an information dissemination problem in which one vertex in a graph, called the originator,
+    must distribute a message to all other vertices by placing a series of calls along the edges of the graph.
+    Once informed, other vertices aid the originator in distributing the message.
+    The broadcasting must be completed as quickly as possible subject to the following constraints:
+    - Each call requires one unit of time.
+    - A vertex can only participate in one call per unit of time.
+    - Each call only involves two adjacent vertices: a sender and a receiver.
+    The minimum broadcast time of a vertex is defined as the minimum amount of time required to complete
+    broadcasting starting from the originator.
+    The broadcast time of a graph is the maximum time required to broadcast from any vertex in the graph [1]_.
```

Comment:
I think saying "maximum over all nodes" helps make the term "maximum" and "required to broadcast" more clear.  Maybe something like:
```suggestion
    The broadcast time of a graph is the maximum over all nodes of the
    minimum broadcast time from that node [1]_.
```

Also, can you wrap the doc_strings so they have at most about 80 chars? The linter checks that for code, but not for doc_strings. But it is better if they stay about that same width.

Reply:
done, I have edited a lot of the docstrings to fit this formatting of 80 chars. (I am presuming the tabs for the indents also count right?). Please let me know if I missed any lines for this.